server/server.go

```diff
@@ -505,11 +505,29 @@ func (s *Server) configureAccounts() error {
 
 	// Set the system account if it was configured.
 	if opts.SystemAccount != _EMPTY_ {
-		// Lock may be acquired in lookupAccount, so release to call lookupAccount.
+		var oldAcc *Account
+		var resetCh chan struct{}
+		if s.sys != nil {
+			oldAcc = s.sys.account
+			resetCh = s.sys.resetCh
+		}
+		// Lock may be acquired in lookupAccount and addSystemAccountExports, so release to call lookupAccount.
 		s.mu.Unlock()
-		_, err := s.lookupAccount(opts.SystemAccount)
+		acc, err := s.lookupAccount(opts.SystemAccount)
+		if err == nil && acc != oldAcc {
+			s.addSystemAccountExports(acc)
+			// can't hold the lock as go routine reading it may be waiting for lock as well
+			if resetCh != nil {
+				resetCh <- struct{}{}
```

Comment:
But I don't understand, you notify here, but we set s.sys.account = acc after reacquiring the lock. So there is a chance that the internal send loop "goto RESET" before the system account is updated, no?

Reply:
Yes you are right. will go back to two relock.